realm/realm-library/src/main/java/io/realm/internal/Property.java

```diff
@@ -33,17 +33,18 @@
     private static final long nativeFinalizerPtr = nativeGetFinalizerPtr();
 
     Property(String name, RealmFieldType type, boolean isPrimary, boolean isIndexed, boolean isRequired) {
-        this.nativePtr = nativeCreateProperty(name, type.getNativeValue(), isPrimary, isIndexed, !isRequired);
+        this.nativePtr = nativeCreatePersistedProperty(name, type.getNativeValue(), isPrimary, isIndexed, !isRequired);
```

Comment:
Object Store has validation. It will throw later anyway. Also, it is something should never happen to user.